### PR TITLE
create-app: fix accidental runtime package dependencies

### LINKED ISSUE #4201: npx @backstage/create-app not working after recent release
## Expected Behavior

`npx @backstage/create-app` should work and scaffold a new Backstage app.

## Current Behavior

It throws an error

```
▶ npx @backstage/create-app

npx: installed 67 in 9.101s
Cannot find module '@backstage/backend-common/package.json'
Require stack:
- /Users/himanshum/.npm/_npx/97126/lib/node_modules/@backstage/create-app/dist/index.cjs.js
- /Users/himanshum/.npm/_npx/97126/lib/node_modules/@backstage/create-app/bin/backstage-create-app
```

## Possible Solution

Is it because of the first release of `@backstage/create-app` with 1.x.x version?

For users: You can try out an older version of create-app script `npx @backstage/create-app@0.3.5`.

## Steps to Reproduce

Run `npx @backstage/create-app`

## Context

Also reported couple times on Discord today
https://discord.com/channels/687207715902193673/687235481154617364/801832402526666762
https://discord.com/channels/687207715902193673/687235481154617364/801849838945173514

### PR DESCRIPTION
**Pull request by Rugvip**
Fixes `@backstage/create-app`, which currently throws the following:

```
Cannot find module '@backstage/backend-common/package.json'
```

Fixes #4201

### CODE REVIEW

**Review by benjdlambert — APPROVED** [review]

**Comment by Rugvip** [comment]
Still haven't figured out how it happened yet, but the last release also bumped create-app to `1.0.0`. I've unpublished that version and rolling back the version here as well

**Comment by OrkoHunter** [comment]
> Still haven't figured out how it happened yet, but the last release also bumped create-app to `1.0.0`. I've unpublished that version and rolling back the version here as well

Yeah, bug with changeset, it seems.

**Review by freben — APPROVED** [review]
Good find... Sheesh this was a nasty onr

File: .changeset/lemon-seahorses-shout.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/create-app': patch
```
[the remark this answers is not in the document]
Comment (Rugvip) [reply .changeset/lemon-seahorses-shout.md:2]: This, only breaking changes to create-app would be changes to the CLI

[the remark this answers is not in the document]
Comment (freben) [reply .changeset/lemon-seahorses-shout.md:2]: Yeah just joking, should be `unmajor`

**Comment by IvanMelzi** [comment]
I am getting the next error: 

`Argument of type 'PluginEnvironment' is not assignable to parameter of type 'CatalogEnvironment'.
  The types returned by 'reader.readTree(...)' are incompatible between these types.`

Node: 14.15.1
npm: 6.14.8

**Comment by Rugvip** [comment]
@IvanMelzi This fix not quite out yet, but will be soon. In addition to having the above bug the version of `create-app` that was published was also incorrectly version as `1.0.0`, so we unpublished it from NPM.

**Pull request merged**